sklearn/linear_model/tests/test_quantile.py

```diff
@@ -250,3 +252,34 @@ def test_linprog_failure():
     msg = "Linear programming for QuantileRegressor did not succeed."
     with pytest.warns(ConvergenceWarning, match=msg):
         reg.fit(X, y)
+
+
+@pytest.mark.parametrize("fit_intercept", [True, False])
+@pytest.mark.parametrize("format", ["csc"])
+def test_sparse_input(fit_intercept, format):
+    rng = np.random.RandomState(42)
+    n = 400
+    dim = 2
+    X = sparse.rand(n, dim, format=format, density=0.5, random_state=rng)
+    beta = rng.rand(dim)
+    y = X * beta[:, np.newaxis]
+    reg = QuantileRegressor(alpha=0, fit_intercept=fit_intercept)
+    reg.fit(X, y.ravel())
+    assert_array_almost_equal(beta, reg.coef_ + reg.intercept_)
+
+    assert_array_almost_equal(reg.predict(X) - y.ravel(), 0)
```

Comment:
```suggestion
@pytest.mark.parametrize("sparse_format", [sparse.csc_matrix, sparse.csr_matrix, sparse.coo_matrix])
@pytest.mark.parametrize("solver", [XXX])
@pytest.mark.parametrize("fit_intercept", [True, False])
def test_sparse_input(sparse_format, solver, fit_intercept):
    """Test that sparse and dense X give same results."""
    X, y = make_regression(n_samples=100, n_features=20, random_state=0, noise=1.0)
    X_sparse = sparse_format(X)
    alpha = 1e-4
    quant_dense = QuantileRegressor(alpha=alpha, fit_intercept=fit_intercept).fit(X, y)
    quant_sparse = QuantileRegressor(alpha=alpha, fit_intercept=fit_intercept, solver=solver).fit(X_sparse, y)
    assert_allclose(quant_sparse.coef_, quant_dense.coef_)
    if fit_intercept:
        assert quant_sparse.intercept_ == approx(quant_dense.intercept_)
        # check that we still predict fraction
        assert np.mean(y < quant_sparse.predict(X_sparse)) == approx(0.5)
```
This test should compare sparse to dense, much like `test_quantile_equals_huber_for_low_epsilon` compares quantile regression to huber regression.

Reply:
Changed the strict equality to 0.5 into a loose condition as- mean needs to be between 0.45 to 0.55.